onnxscript/rewriter/generic_pattern.py

```diff
@@ -294,6 +294,78 @@ def make_node(
         return proto
 
 
+class PatternMatchResult:
```

Comment:
Is this a user facing class? Can it be private? Or can the whole module be private?

Reply:
Yes it is part of the public API, when the user writes the validation function.

Reply:
If so I would be very careful in designing the fields and making sure we are able to evolve them

Reply:
What do you mean by 'able to evolve them'?

Reply:
So if we add or remove fields in the initializer, older usages won't break.

Reply:
I'm still not sure about you mean. You obviously don't break anything by adding more fields. It is very likely to break if an attribute is removed.